src/main/java/bong/Bong.java

```diff
@@ -0,0 +1,90 @@
+package bong;
+
+import javafx.application.Application;
+
+import bong.command.Command;
+import bong.command.CommandResult;
+import bong.task.TaskList;
+import bong.utils.BongException;
+import bong.utils.Parser;
+import bong.utils.Storage;
+
+/**
+ * The {@code Bong} class is the main entry point of the application.
+ * It initializes the necessary components and starts the main program loop
+ * to process user commands.
+ */
+public class Bong {
+    private static final String DEFAULT_FILE_PATH = "./data/bong.txt";
+    private final Storage storage;
+    private TaskList tasks;
+    private final Parser parser;
+
+    /**
+     * Constructs a new {@code Bong} instance.
+     * Initializes the storage, parser, and loads tasks from the storage file.
+     * If the storage file cannot be loaded, an empty task list is initialized.
+     */
+    public Bong() {
+        storage = new Storage(DEFAULT_FILE_PATH);
+        parser = new Parser();
+        loadTasks();
+    }
+
+    /**
+     * Loads tasks from storage or initializes an empty task list if loading fails.
+     */
+    private void loadTasks() {
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (BongException e) {
+            tasks = new TaskList();
+        }
+
+        assert tasks != null : "TaskList should not be null after initialization";
+    }
+
+    /**
+     * Processes a user input, executes the corresponding command,
+     * and returns the result as a response.
+     *
+     * @param userInput The user input string.
+     * @return The feedback message to the user after executing the command.
+     */
+    public String getResponse(String userInput) {
```

Comment:
Nice job implementing the core logic for handling user commands! One minor suggestion would be to include more specific exception handling for commands that might fail due to user input errors. This way, you could provide more detailed feedback to the user instead of the generic 'An error occurred.' message.